[CECO-2145] Use `datadog.kubelet.useApiServer` for GKE Autopilot (#29199)
DataDog/documentation: content/en/containers/kubernetes/distributions.md

GKE Autopilot requires some configuration, shown below.

Datadog recommends that you specify resource limits for the Agent container. Autopilot sets a relatively low default limit (50m CPU, 100Mi memory) that may lead the Agent container to quickly OOMKill depending on your environment. If applicable, also specify resource limits for the Trace Agent, Process Agent and System-Probe containers. Additionally, you may wish to create a priority class for the Agent to ensure it is scheduled.

> Starting with Agent `7.65.0+` and version `3.113.0+` of the Helm chart, we also recommend using `datadog.kubelet.useApiServer` for the Agent to query the pod list from the API server instead of the [deprecated read-only kubelet port][12].

Reviewer: ```suggestion
Starting with Agent `7.65.0+` and version `3.113.0+` of the Helm chart, Datadog recommends using `datadog.kubelet.useApiServer` for the Agent to query the pod list from the API server. Avoid using the [deprecated read-only kubelet port][12].
```
